File: desafio-fullcycle/src/domain/product/service/product.service.ts
import { Product } from "../entity/product";

export class ProductService {
  public static increasePrice(products: Product[], percentage: number): void {
    products.forEach(product => {
      const newPrice = product.price * (1 + percentage / 100);
      product.changePrice(newPrice);
      return product;
    });
  }
}

File: entidades-exemplo/src/main/java/com/fredsonchaves07/domain/product/service/ProductService.java
package com.fredsonchaves07.domain.product.service;

import com.fredsonchaves07.domain.product.entity.Product;

import java.util.List;

public class ProductService {

    public static void increasePrices(List<Product> products, int percent) {
        products.forEach(product -> product.changePrice((product.getPrice() * percent) / 100 + product.getPrice()));
    }
}
